ssl/t1_lib.c

```diff
@@ -1512,7 +1761,9 @@ int tls12_check_peer_sigalg(SSL_CONNECTION *s, uint16_t sig, EVP_PKEY *pkey)
         return 0;
     }
     /* Check the sigalg is consistent with the key OID */
-    if (!ssl_cert_lookup_by_nid(EVP_PKEY_get_id(pkey), &cidx)
+    if (!ssl_cert_lookup_by_nid(
+                 (pkeyid==EVP_PKEY_RSA_PSS)?EVP_PKEY_get_id(pkey):pkeyid, &cidx,
```

Comment:
nit: missing spaces around `==` and `?` and `:`

Reply:
So changed in 0ac814f8247d1a5f157eec2b62bce0000c3a2d1e.